app/features/worksheet/tools.py

```diff
@@ -0,0 +1,278 @@
+            "embedding_model": SentenceTransformer('all-MiniLM-L6-v2'),
+            "prompt_summary": read_text_file("prompts/worksheet_prompt_summary.txt"),
+            "parser_summary": JsonOutputParser(pydantic_object=SummaryFormat),
+            "prompt_fill_in_blank": read_text_file('prompts/worksheet_prompt_fill_in_blank.txt'),
+            "parser_fill_in_blank": JsonOutputParser(pydantic_object = FillinblankQuestionFormat),
+            "prompt_multiple_choice": read_text_file("prompts/worksheet_prompt_multiple_choice.txt"),
+            "parser_multiple_choice": JsonOutputParser(pydantic_object = MultipleChoiceQuestionFormat),
+            "prompt_open_ended": read_text_file("prompts/worksheet_prompt_open_ended.txt"),
+            "parser_open_ended": JsonOutputParser(pydantic_object = OpenEndedQuestionFormat),
+            "prompt_true_false": read_text_file("prompts/worksheet_prompt_true_false.txt"),
+            "parser_true_false": JsonOutputParser(pydantic_object = TrueFalseQuestionFormat)
+        }
+        
+        self.model = model or default_config["model"]
+        self.embedding_model = embedding_model or default_config["embedding_model"]
+        self.prompt_summary = prompt_summary or default_config["prompt_summary"]
+        self.parser_summary = parser_summary or default_config["parser_summary"]
+        self.prompt_fill_in_blank = prompt_fill_in_blank or default_config["prompt_fill_in_blank"]
+        self.parser_fill_in_blank = parser_fill_in_blank or default_config["parser_fill_in_blank"]
+        self.prompt_multiple_choice = prompt_multiple_choice or default_config["prompt_multiple_choice"]
+        self.parser_multiple_choice = parser_multiple_choice or default_config["parser_multiple_choice"]
+        self.prompt_open_ended = prompt_open_ended or default_config["prompt_open_ended"]
+        self.parser_open_ended = parser_open_ended or default_config["parser_open_ended"]
+        self.prompt_true_false = prompt_true_false or default_config["prompt_true_false"]
+        self.parser_true_false = parser_true_false or default_config["parser_true_false"]
+        
+        self.topic = topic
+        self.grade_level = grade_level
+        self.verbose = verbose
+    
+    def create_worksheets(self, num_worksheets: int = 1, num_fill_in_blank: int = 1, num_multiple_choice: int = 1, num_open_ended: int = 1, num_true_false: int = 1) -> List[str]:
+        ### Limit just for testing
+        if num_worksheets > 10:
+            return {"message": "error", "data": "Number of Worksheets cannot exceed 10"}
+        
+        summary_obj = Summary(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_summary, self.parser_summary, self.verbose)
+        fill_in_the_blank_obj = FillInTheBlankQuestion(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_fill_in_blank, self.parser_fill_in_blank, self.verbose)
```

Comment:
Use a Pydantic Schema for reuse the args as well as separating this in another function for not repeating the same parts of the code.